packages/dds/merge-tree/src/client.ts

```diff
@@ -346,6 +350,49 @@ export class Client {
         return this.mergeTree.getMarkerFromId(id);
     }
 
+    /**
+     * Revert an op
+     */
+    public rollback?(op: any, localOpMetadata: unknown) {
+        if (op.type !== MergeTreeDeltaType.INSERT) {
+            throw new Error("Unsupported op type for rollback");
+        }
+
+        const lastList = this.mergeTree.pendingSegments?.prev;
+        if (!lastList) {
+            throw new Error("No pending segments");
+        }
+        ListRemoveEntry(lastList);
+        const pendingSegmentGroup = lastList.data;
+        if (pendingSegmentGroup === undefined || pendingSegmentGroup !== localOpMetadata) {
+            throw new Error("Rollback op doesn't match last edit");
+        }
+        let length = 0;
+        for (const segment of pendingSegmentGroup.segments) {
+            const segmentSegmentGroup = segment.segmentGroups.dequeue();
+            assert(segmentSegmentGroup === pendingSegmentGroup, "Unexpected segmentGroup in segment");
+            if (TextSegment.is(segment)) {
+                length += segment.text.length;
+            } else if (Marker.is(segment)) {
+                length += 1;
```

Comment:
I'm not sure I should be assuming a marker is always length 1 in case that changes but I don't know how else to figure it out.